src/material/core/focus-indicators/_focus-indicators.scss

```diff
@@ -76,26 +111,46 @@
   }
 }
 
-/// Mixin that sets the color of the focus indicators.
+/// Mixin that customizes focus indicators. This mixin may be included
+/// multiple times throughout your app.
 ///
-/// @param {color|map} $theme-or-color
-///   If theme, focus indicators are set to the primary color of the theme. If
-///   color, focus indicators are set to that color.
+/// @param {map} $config
+///   If given, customizes focus indicators with the configuration. The
+///   configuration map accepts border-style, border-color, border-width,
+///   border-radius, and margin. See the SassDoc annotation for
+///   mat-strong-focus-indicator for more on each parameter.
 ///
-/// @example
-///   .demo-dark-theme {
-///     @include mat-strong-focus-indicators-theme($dark-theme-map);
-///   }
+/// @param {boolean} $cascade
+///   If true, the focus indicator customizations will apply to the
+///   entire element subtree. Otherwise, they only apply to the selector
+///   hosting the mixin. Defaults to true.
 ///
 /// @example
-///   .demo-red-theme {
-///     @include mat-strong-focus-indicators-theme(#f00);
+///   .my-component {
+///     @include mat-strong-focus-indicator-customize((
+///       border-radius: 50%,
+///       color: red,
+///     ));
 ///   }
-@mixin mat-strong-focus-indicators-theme($theme-or-color) {
-  .mat-focus-indicator::before {
-    border-color: if(
-      type-of($theme-or-color) == 'map',
-      mat-color(map_get($theme-or-color, primary)),
-      $theme-or-color);
+@mixin mat-strong-focus-indicator-customize($config: (), $cascade: true) {
```

Comment:
The motivation behind the `$cascade` parameter is that sometimes it makes sense for focus indicator styles to cascade to an entire element subtree (e.g. inverting a focus indicator's color on a dark background), and sometimes it makes sense for focus indicator styles to only be applied to the selector hosting the mixin (e.g. the use cases within the `mat-strong-focus-indicator` mixin). I know that we don't love multiple mixin parameters... but it doesn't really make sense to add `$cascade` to the `$config` parameter because it's a different type of parameter.

Also curious if we can think of other ways to solve this problem.

Reply:
The way this is written, how would you use `cascade: false`? It looks like the class would have to be on the same element as `.mat-focus-indicator`, which users wouldn't have access to. 

Also, `cascade` could just be another option in `$config` that we filter out from rendering.

Reply:
Suppose you write:

```
.mat-card {
  @include mat-strong-focus-indicator-customize((border-color: red));
}
```

The reasoning behind `cascade` is that sometimes you want to target `.mat-card` and sometimes you want to target all of the focusable components inside `.mat-card`. AFAIK, there's no way to do that without either what I have here or writing `.mat-focus-indicator::before, &.mat-focus-indicator::before { ... }`.

> It looks like the class would have to be on the same element as .mat-focus-indicator, which users wouldn't have access to.

Why wouldn't users have access to this element? In the example above, the class is on the `.mat-card`, which users have access to. I know that for components like checkbox the focus indicator is somewhat buried inside the component template, is that what you mean? This doesn't seem like a new problem though, don't we have the same problem today when users want to style things like ripples?

> Also, cascade could just be another option in $config that we filter out from rendering.

I'm not sure what you mean from "filter out from rendering", sorry.

Reply:
Yeah, I meant for components where the focus indicator is within the template. I'm not a fan of adding an API (`cascade` here) that would only work for a subset of the components; doing `cascade: false` just doesn't work for those components. Maybe it doesn't matter, though, if all of those components don't allow other form controls inside of them (like checkbox and radio). 

For the `cascade` param, I mean just making it part of the map and then ignoring it in `@each $key in $config-keys`.

Reply:
Makes sense. See the other thread where I suggest two mixins instead of `cascade`: customize `-self` and customize `-children` mixins. The `-self` mixin uses `&.mat-focus-indicator` and the `-children` mixin uses `.mat-focus-indicator` (and thus cascades).

> I'm not a fan of adding an API (cascade here) that would only work for a subset of the components

With the two mixin approach, I think it's fine that the `mat-strong-focus-indicator-children` would only work for a subset of components. Like you said, it wouldn't work for any components that don't allow nested focusable elements (which is expected).